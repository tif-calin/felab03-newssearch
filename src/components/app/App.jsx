import React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect
} from 'react-router-dom';
import { NewsSearch } from '../../containers/NewsSearch';

export default function App() {
  return <main className="App">
    <Router>
      <Switch>

        <Route exact path="/" render={() => (
          <Redirect to="/news"/>
        )}/>

        <Route path="/news" render={routerProps => 
          <NewsSearch {...routerProps}/>
        }/>

        <Route path="/news/:id" render={routerProps => 
          <NewsSearch {...routerProps}/>
        }/>

        <Redirect to="/"/>

      </Switch>
    </Router>
  </main>;
}
