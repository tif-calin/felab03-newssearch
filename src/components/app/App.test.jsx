import React from 'react';
import { 
  render, 
  cleanup,
  fireEvent,
  screen,
  waitFor
} from '@testing-library/react';
import App from './App';

describe('App component', () => {
  afterEach(() => cleanup());

  it('renders App', async () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('test search functionality', async () => {
    render(<App />);

    const searchInput = screen.getByDisplayValue('kahnawake');
    
    // search for florida 
    fireEvent.change(searchInput, { target: { value: 'florida' } });
    
    waitFor(async () => {
      const newsList = await screen.findByRole('list');
      expect(newsList).not.toBeEmptyDOMElement();
      expect(newsList).toContainHTML('</li>');

      const newsItem = await newsList.findByTestId('article');
      expect(newsItem).not.toBeEmptyDOMElement();
    }, 9000);

  });

});
