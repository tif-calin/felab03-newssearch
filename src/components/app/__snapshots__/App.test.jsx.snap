// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <main
    class="App"
  >
    <div
      class="NewsSearch"
    >
      <div
        class="Search"
      >
        <input
          type="text"
          value="kahnawake"
        />
      </div>
      <span>
        loading
      </span>
    </div>
  </main>
</DocumentFragment>
`;
